### DISCUSSION TITLE
Refresh interval per panel

### DISCUSSION THREAD

**Discussion by lacion** [grafana/grafana#36965] [Dashboards, Panels & Visualizations]
it would be useful specially now that there is elastic search and cloud watch backends to be able to set a per graph refresh i would like to have dashboards that will show a general overview some times over a week running pretty hard queries over in elastic search, but also showing some graph from other sources (like graphite or others db's) where i would like near realtime updates.

having real time updates on ES and cloud watch is really heavy on ES resources and cost's on aws but being able to override the refresh rate per graph would be really helpful.

**Comment by gbrian** [comment]
+1
What's the way for helping on this ?
UPDATE: ok, found http://docs.grafana.org/project/cla/

**Comment by utkarshcmu** [comment]
How about per row refresh rate instead of per graph?

**Comment by gbrian** [comment]
First version done:
Now refresh is by panel and you can set the value on each panel, at row level or dashboard level.
Basically what it does, at dashboard refresh, is to assign a refresh rate to each panel. 

Main interval is calculated to be the greatest common divisor and that's all.

Pending last testing and create the pull request.

![image](https://cloud.githubusercontent.com/assets/767180/12228302/96b1f93a-b839-11e5-8214-ab0c4f64b65b.png)
![image](https://cloud.githubusercontent.com/assets/767180/12228309/a6e50da6-b839-11e5-99d7-835f68018065.png)

![image](https://cloud.githubusercontent.com/assets/767180/12228255/ed948e12-b838-11e5-8e7d-08f5d3a2a882.png)

UPDATE: Ops, changing "Add refresh shift" by "Override refresh time"

**Comment by mtanda** [comment]
This is useful for CloudWatch to reduce cost.

**Comment by gbrian** [comment]
I've had to close pull request as there are many changes on master to keep it updated (test were failing as well). I'll try to implement again over latest version unless @torkelo you already have started this implementation.

**Comment by jsdt** [comment]
+1

**Comment by Codelica** [comment]
I think this would be very useful for mixed content dashboards.  I'd love to be able to mix some realtime-ish charts in with longer term trends.  Any chance this could be revived ?

**Reply by celestialorb** [reply]
This is my desired use case as well; that and for reducing the refresh rate on panels that have heavy queries.

**Comment by arianitu** [comment]
+1, I really assumed Grafana would support something like this.

**Comment by leafonsword** [comment]
Hope support row level, panel level refreshing!

**Comment by hatzlhoffer** [comment]
+1 this would be very helpful!

**Comment by kccheung** [comment]
+1

**Comment by dooe** [comment]
+1

**Comment by colmobrien84** [comment]
+1 this would greatly add to my Grafana <3

**Comment by pedrohtex** [comment]
+1

**Comment by scaarup** [comment]
+1

**Comment by unixfox** [comment]
Is it possible to stop posting the useless `+1` and just upvote the first comment on this issue if you are interested into this feature?

**Comment by arnolde** [comment]
I'm surprised this was implemented 3 years ago but not released yet... Please upvote more :-)

**Comment by Edu-GM** [comment]
+1

**Comment by arnolde** [comment]
@Edu-GM Please don't write +1, that is no help. Click the thumbs-up on the very first message in this issue, that will effectively upvote this issue.

**Comment by Knot3n** [comment]
Please release this feature .. we need it to have a bit less load on the server.

**Comment by gbrian** [comment]
Hi everyone,
Did new attempt to implement this feature. The PR is just a preview for sharing the approach and grab some comments.
Please if you have the time check it and share your thoughts
https://github.com/grafana/grafana/pull/16384

**Comment by gbrian** [comment]
Ok, done tested and working fine! (https://github.com/grafana/grafana/pull/16384)
hope they'll like the implementation 😅 @daniellee @torkelo 
![grafana_panel_refresh](https://user-images.githubusercontent.com/767180/55616207-9862d880-5791-11e9-8952-14e3298ba195.gif)

![image](https://user-images.githubusercontent.com/767180/55616309-d3650c00-5791-11e9-89e2-7206027d8514.png)

**UPDATE**:
Now realized about the migration to react 😮 so need to re-implement for react components...ok
#challengeaccepted

**Comment by gbrian** [comment]
Ok, guess I have it done for the React panels 😓 
![grafana_panel2](https://user-images.githubusercontent.com/767180/55622680-2c886c00-57a1-11e9-9b1d-de66995118fe.gif)

**Comment by MissMusic** [comment]
I need this so much!

**Comment by em0il** [comment]
can someone tell me how I can achieve different refresh intervals (by own coding or where can I find the panel from above)

**Comment by mario-donnarumma** [comment]
I think this feature would be great. Definitely need this.

**Comment by davide-acanfora** [comment]
This would be a great feature for dashboards with multiple datasources :+1:

**Comment by dbu748** [comment]
I would love to have this feature ! :+1:

**Comment by amyiludur** [comment]
This would be an amazing feature and its the only reason why I'm hesitant to start using this platform +1

**Comment by szerwi** [comment]
Is there any update about this future? I'd like to see it in Grafana.

**Comment by zachthedev** [comment]
@torkelo Any update?

**Comment by beingyash** [comment]
@torkelo Is this feature request anywhere on the current roadmap?

**Comment by torkelo** [comment]
no

**Comment by dnfm** [comment]
Why would it be?  It's not like a tonne of people haven't wanted it for nearly 5 years now!

... oh wait..

**Comment by beingyash** [comment]
> no

Do you have some defined criteria to consider feature requests, so as to get this one qualified?

**Comment by szerwi** [comment]
@torkelo so maybe at least some protection that if previous refresh is not finished, it won't start another one. I have a few panels which took long to load and when I set refresh rate to 10s it is impossible for them to load, I need to manually change refresh rate to 1 minute so they load properly.

**Comment by gbrian** [comment]
Hi @torkelo ,
First things first, thanks for taking the time to answer comments in this thread as I can perfectly understand the reason for Grafana- and many other open source projects- to keep focus.

So this is a thought for all people here willing to have this, and other features, in Grafana:

What if there's a fork adding this functionalities and offering the pull request to the main repository? Do you know if something like this exists?

 * New features/bug fixes will be deployed following Grafana's guideline
 * New features will be discussed/agreed with repository owners to ensure its adoption by main repository in the future.
 * All developments will have each own branch
 * There will be a main branch with all developments or option to build your own set
 * Pull requests will be updated with new changes coming from main repository
 * Fork will use platform like https://opencollective.com/ to sustain developments

Comments @torkelo ?, on your time.

**Comment by torkelo** [comment]
We accept pull requests for popular feature request if the implementation is sound and the added complexity is low enough compared to the added value

**Comment by hanahas** [comment]
👍

**Comment by kj187** [comment]
👍

**Comment by kadamik** [comment]
👍

**Comment by SkateWarp** [comment]
👍

**Comment by gbrian** [comment]
Time flies... https://github.com/grafana/grafana/discussions/36965#discussioncomment-1028355 😅 looks like this still hot topic

**Comment by scaarup** [comment]
3 years since I +1'ed :)

**Comment by jtfidje** [comment]
Yeah this would be awesome !

**Comment by cyraid** [comment]
Does moving to discussion mean dusting it under the rug?

**Comment by discarn8** [comment]
+1 & **please**?

**Comment by melvanderwal** [comment]
I would also really appreciate this addition.   I commonly have dashboards wtih a mix of current and historic information.   Current example:  solar system monitoring - gauges showing current output from a near-realtime JSON data source, and historic trend time series graphs from an InfluxDB cloud data source.  InfluxDB Cloud charges for transactions, so I'd prefer to be able to refresh the current data far more frequently than the time-series data.

**Comment by GollaSaiVenkatesh** [comment]
yeah this will be an awesome feature to have.

**Comment by maxdd** [comment]
+1

**Comment by AnouarKh** [comment]
any news on this feature :) ?

**Comment by milinpatel13298** [comment]
This would be a very resourceful feature. Perhaps, an additional field in "Query options"?

**Comment by cyberthom42** [comment]
I am confused. Hast there been a workaround or Implementation proposed?

**Comment by zcattacz** [comment]
We need this feature when near-realtime status and history show on the same page.
An alternative way to support this is to allow referencing panel data from different dashboards as data source, instead of from the same dashboard, meanwhile keep the source_panel_refresh_interval as max refresh interval for that data.
Another alternative way is to support sort of framed dashboard by putting sub-dashboards on a framed dashboard.
Both can circumvent this issue.

**Comment by cyberthom42** [comment]
So it seems that the user individually need to overcome this issue. I seems that grafana team will not implement this important feature. I will try to implement this by creating several individual i-frames. But to be honest, this is not what I would expect from such a project...

**Comment by piecler** [comment]
This feature is clearly most wanted by a lot of people.
And there is even an implementation by @gbrian 
I have also historical data on my dashboard which may take several seconds to update, yet I also need real time data to update every second.
This feature is neither complicated nor breaks anything.
Why is it totally ignored by the devs?
Please tell us _something_!
Can we expect this feature or do we have to patch this on every update ourselves? Would be annoying as hell, but I really need this.
Thank you.

**Comment by gbrian** [comment]
@torkelo, @daniellee I'd love to try again, worth it trying? Any recommendations? :)

**Reply by celestialorb** [reply]
I'd also like to try implementing this, just might take a bit to familiarize myself with the codebase.

**Reply by gbrian** [reply]
Great! I'm setting up a coding env we can share and meet to discuss. I'll share once ready. fine?

**Reply by torkelo** [reply]
This would need to be implemented in the new dashboard architecture (scenes)

**Reply by gbrian** [reply]
First take:
<img width="290" alt="image" src="https://user-images.githubusercontent.com/767180/222795448-4dafff71-58e9-4c82-a030-ec67e0bbeb2c.png">

Added `refreshShift` and `currentShift` to Panel. Basically we can skip X refresh events per panel.

In the example dashboard has 5s refresh and `Panel B` has `refreshShhit == 2` refreshing every 10s instead.

<img width="1361" alt="image" src="https://user-images.githubusercontent.com/767180/222795820-68a426ab-fb3f-47a8-8dca-76419e44600b.png">

code implementation:

<img width="654" alt="image" src="https://user-images.githubusercontent.com/767180/222796000-400ef11e-dff1-417e-9558-24e3d2bdddc3.png">

**Reply by gbrian** [reply]
So, is this the right approach for current architecture, @torkelo ? BTW Reading about scenes in the code... hints are welcome

**Reply by BrunoBlanes** [reply]
This is an approach I'd say, in my opinion a lazy one though, however a great start. Ideally, I think what people would expect is to be a menu somewhere for each panel for you to select from the various intervals already available.

**Reply by gbrian** [reply]
Thanks. Yes v0.0, wanted something that works and makes minimum impact possible, this change is quite easy to track. Probably we can change the concept to a "slow down refresh" and have some indication in then panels like `Panel Titel (🐢x2)` (other something muuuch better) meaning this panel has a "x2" slow refresh from the main refresh.

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This discussion was converted into an issue, which is available here: https://github.com/grafana/grafana/issues/73434
